### ISSUE TITLE
two models with identical parameters give different prediction AUC on the same dataset

### ISSUE THREAD

**Issue by stb-f5c** [dmlc/xgboost#8654]
Hi, I try to replicate a registered XGBoost model A on databricks, and I get model B, with identical following parameters from tuning:

alpha, colsample_bytree, gamma, lambda, learning_rate, max_delta_step, max_depth, min_child_weight,best_iteration

I notice that the two models have slightly different feature importance values.
When I predict on the same test dataset, two models give me two different model AUC on test data, 0.8 vs 0.75
I am curious what can be the reasons behind the two different predictions?
Did I overlook some other parameters here?

**Comment by trivialfis** [comment]
Could you please share some more information, what's the versions of XGBoost used to train these models? What's the number of workers used to train these models? How did you train the model, like which interface are you using?

**Comment by stb-f5c** [comment]
@trivialfis thank you for the reply.
what are the version of XGBoost is 1.6.1
python version 3.8.10

I train the model using Microsoft Azure Databricks, the Databricks Runtim version is 10.4 LTS ML (10.4.x-cpu-ml-scala2.12) . the cluster has 2-4 workers.
, I use hyperopt for tuning. and I register my model using mlflow (version 1.28).

**Comment by trivialfis** [comment]
hmm .. I can't really debug it:
- tree models are sensitive to data (hence known to over-fit). One split difference in an early tree can lead to all the following trees being different.
- AUC is not a particularly robust metric in a distributed environment, as documented in the auc metric from xgboost's doc.
- xgboost might change training results if the number of workers is different between two training sessions.

**Comment by stb-f5c** [comment]
Thanks again for the update. Can I ask why AUC is not a robust metric in this case, and if you have any advice for a walk-around or alternative to AUC as a metric here?

**Comment by stb-f5c** [comment]
Hi again,
I try to replicate the same result as the model with 80% AUC in test data. I use the same training data, and train the same way, and register a model with the same hyperparamters listed above from the 80% AUC model. And again the newly created model only gives 75% AUC on the same test data. Not sure what else, apart from the "unstability" from distrubted environment and randomness, that can explain the difference here. The 5% difference is quite substaintial.. Appreciate any comment/tips, thanks!

**Comment by trivialfis** [comment]
any chance you can show me how to reproduce it?

**Comment by stb-f5c** [comment]
Unfortunately it is not possible due to regulatory issue, but thanks for asking and the help here. I close the issue for now.
